pkg/edgehub/controller_test.go

```diff
@@ -0,0 +1,885 @@
+		msgID  string
+	}{
+		{"TestEdgeHubController_addKeepChannel: Positive Test Case", EdgeHubController{
+			syncKeeper: make(map[string]chan model.Message),
+		},
+			"test"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ehc := &EdgeHubController{
+				context:    tt.fields.context,
+				chClient:   tt.fields.chClient,
+				config:     tt.fields.config,
+				stopChan:   tt.fields.stopChan,
+				syncKeeper: tt.fields.syncKeeper,
+				keeperLock: tt.fields.keeperLock,
+			}
+			got := ehc.addKeepChannel(tt.msgID)
+			if !reflect.DeepEqual(ehc.syncKeeper[tt.msgID], got) {
+				t.Errorf("EdgeHubController.addKeepChannel() = %v, want %v", got, ehc.syncKeeper[tt.msgID])
+			}
+		})
+	}
+}
+
+//TestDeleteKeepChannel() tests the deletion of channel in the syncKeeper
+func TestDeleteKeepChannel(t *testing.T) {
+	tests := []struct {
+		name   string
+		fields EdgeHubController
+		msgID  string
+	}{
+		{"TestEdgeHubController_deleteKeepChannel: Positive Test Case",
+			EdgeHubController{
+				syncKeeper: make(map[string]chan model.Message),
+			},
+			"test"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ehc := &EdgeHubController{
+				stopChan:   tt.fields.stopChan,
+				syncKeeper: tt.fields.syncKeeper,
+				keeperLock: tt.fields.keeperLock,
+			}
+			ehc.addKeepChannel(tt.msgID)
+			ehc.deleteKeepChannel(tt.msgID)
+			if _, exist := ehc.syncKeeper[tt.msgID]; exist {
+				t.Errorf("TestEdgeHubController_deleteKeepChannel = %v, want %v", ehc.syncKeeper[tt.msgID], nil)
+			}
+		})
+	}
+}
+
+//TestIsSyncResponse() tests whether there exists a channel with the given message_id in the syncKeeper
+func TestIsSyncResponse(t *testing.T) {
+	tests := []struct {
+		name   string
+		fields EdgeHubController
+		msgID  string
+		want   bool
+	}{
+		{"TestEdgeHubController_isSyncResponse: true case",
+			EdgeHubController{
+				syncKeeper: make(map[string]chan model.Message),
+			}, "test",
+			true,
+		},
+		{"TestEdgeHubController_isSyncResponse: false case",
+			EdgeHubController{
+				syncKeeper: make(map[string]chan model.Message),
+			}, "",
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ehc := &EdgeHubController{
+				context:    tt.fields.context,
+				chClient:   tt.fields.chClient,
+				config:     tt.fields.config,
+				stopChan:   tt.fields.stopChan,
+				syncKeeper: tt.fields.syncKeeper,
+				keeperLock: tt.fields.keeperLock,
+			}
+			if tt.want {
+				ehc.addKeepChannel(tt.msgID)
+			}
+			if got := ehc.isSyncResponse(tt.msgID); got != tt.want {
+				t.Errorf("EdgeHubController.isSyncResponse() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+//TestSendToKeepChannel() tests the reception of response in the syncKeep channel
+func TestSendToKeepChannel(t *testing.T) {
+	tests := []struct {
+		name              string
+		fields            EdgeHubController
+		message           *model.Message
+		wantSyncKeeperErr bool
+		wantErr           bool
```

Comment:
wantBool ?? Applicable to all cases below

Reply:
Here wantErr variable is used to signify whether it is a positive test case or not. It basically is used to tell the user whether the test case is expecting an error or not. If wantErr is true, it means that an error is expected in the test case(negative test case), if it is false then it signifies that no error is expected. I feel that wantBool might not convey the message properly and might not be easy to understand.